PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx

```diff
@@ -254,6 +258,44 @@ struct HfCorrelatorLcHadrons {
     corrBinning = {{binsZVtx, binsMultiplicity}, true};
   }
 
+  // Find Leading Particle
+  template <typename TTracks>
+  int findLeadingParticle(TTracks const& tracks)
```

Comment:
`findLeadingParticle` and `findLeadingParticleMcGen` are also duplications of methods in `PWGHF/HFC/TableProducer/correlatorD0Hadrons.cxx`.
Please move these in the same header as well.

Reply:
I have now tried to add it in the same header, but I am not sure if it meets your suggestions.